app/services/process_notify_callback.rb

```diff
@@ -7,7 +7,14 @@ def call(notify_reference:, status:)
       return :not_updated unless same_environment? && reference_request_email? && permanent_failure_status?
       return :not_found unless ApplicationReference.exists?(reference_id)
 
-      ApplicationReference.find_by(id: reference_id).update!(feedback_status: 'email_bounced')
+      reference = ApplicationReference.find_by(id: reference_id)
+      reference.update!(feedback_status: 'email_bounced')
```

Comment:
This makes 2 queries to check/fetch the reference. You could slim this down by using `ApplicationReference.find(reference_id)` and rescuing `ActiveRecord::RecordNotFound` at the end of the method.

Reply:
Makes sense! Done in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1157/commits/b99f0893026bbf26f8fa4aa4d71778f8492a4756.